src/generate.ts

```diff
@@ -0,0 +1,192 @@
+import type {
+	AbstractSqlField,
+	AbstractSqlModel,
+	AbstractSqlTable,
+	InNode,
+	Relationship,
+	RelationshipInternalNode,
+	RelationshipLeafNode,
+} from '@balena/abstract-sql-compiler';
+import {
+	odataNameToSqlName,
+	sqlNameToODataName,
+} from '@balena/odata-to-abstract-sql';
+import { replaceResultTransformer, TemplateTag } from 'common-tags';
+
+type RequiredModelSubset = Pick<
+	AbstractSqlModel,
+	'tables' | 'relationships' | 'synonyms'
+>;
+
+const typeHelpers = `import type { Types } from '@balena/abstract-sql-to-typescript';\n`;
+
+const trimNL = new TemplateTag(
+	replaceResultTransformer(/^[\r\n]*|[\r\n]*$/g, ''),
+);
+
+const modelNameToCamelCaseName = (s: string): string =>
+	s
+		.split(/[ -]/)
+		.map((p) => p[0].toLocaleUpperCase() + p.slice(1))
+		.join('');
```

Comment:
```suggestion
const modelNameToCamelCaseName = (s: string): string =>
    s.replace(/(?:^\w|[A-Z]|\b\w)/g, word => word.toUpperCase())
     .replace(/\s+|-+/g, '');
```

Nit. Possible optimization using regex instead of creating an array and joining it back into a string.

Reply:
I think that would probably work but I would avoid it in this case. Since this is compile time code only and is already very quick (~15-25ms for the generation, reading input/importing dependencies/writing output is the vast majority of time spent in practice) then I'd prefer to pick readable/understandable code rather than optimizations like this - if it was hot runtime code though then it'd be a very different story